solidity/dashboard/src/components/NetworkStatus.jsx

```diff
@@ -1,24 +1,27 @@
-import React, { useContext } from "react"
-import { Web3Context } from "./WithWeb3Context"
+import React from "react"
+import { useWeb3Context } from "./WithWeb3Context"
+import * as Icons from "./Icons"
 
 export const NetworkStatus = () => {
-  const { networkType, provider, error, yourAddress } = useContext(Web3Context)
+  const { networkType, provider, error, yourAddress } = useWeb3Context()
+
+  let status = "disconnected"
+  if (error) {
+    status = "error"
+  } else if (yourAddress && !error && provider !== null) {
+    status = "connected"
+  }
 
   return (
     <div className="network-status flex row center">
-      <div
-        className={`network-indicator ${
-          yourAddress && !error && provider !== null ? "connected" : "error"
-        }`}
+      <Icons.NetworkStatusIndicator
+        className={`network-status__indicator--${status}`}
       />
-      <h5 className="text-grey-50">
-        {!yourAddress && "not connected"}
-        {yourAddress &&
-          !error &&
-          provider !== null &&
-          `connected: ${networkType}`}
-        {yourAddress && error && provider !== null && `wrong network`}
-      </h5>
+      <div className={`network-status__text--${status}`}>
```

Comment:
Is this truly a `div`? Could we get away with no element or a `span`? It doesn't feel block-level at first blush.